test/compliance/UPDATE.test.js

```diff
@@ -49,4 +50,93 @@ describe('UPDATE', () => {
       throw new Error('not supported')
     })
   })
+
+  describe('uniques in deep updates', () => {
+    cds.test(__dirname + '/resources')
+    test('2nd level unique constraints ', async () => {
+      // number must be unique for each book
+
+      await cds.tx(async tx => {
+        await tx.run(DELETE.from(BooksUnique).where({ ID: 1 }))
+        await expect(
+          tx.run(
+            INSERT.into(BooksUnique).entries([
+              {
+                ID: 1,
+                title: 'foo',
+                pages: [
+                  {
+                    ID: 1,
+                    number: 1,
+                  },
+                  {
+                    ID: 2,
+                    number: 1, // unique constraint violation
+                  },
+                ],
+              },
+              {
+                ID: 2,
+                title: 'bar',
+              },
+            ]),
+          ),
+        ).rejects.toBeTruthy()
+      })
+
+      await cds.tx(async tx => {
+        await tx.run(DELETE.from(BooksUnique).where({ ID: 1 }))
+        await tx.run(
+          INSERT.into(BooksUnique).entries([
+            {
+              ID: 1,
+              title: 'foo',
+              pages: [
+                {
+                  ID: 1,
+                  number: 1,
+                },
+                {
+                  ID: 2,
+                  number: 2,
+                },
+              ],
+            },
+          ]),
+        )
+        await tx.run(
+          UPDATE(BooksUnique).data({
+            ID: 1,
+            title: 'foo',
+            pages: [
+              {
+                ID: 3,
+                number: 1,
+              },
+              {
+                ID: 4,
+                number: 2,
+              },
+            ],
+          }),
+        ) // first, old entries are deleted, so no violation
+        await tx.run(
+          UPDATE(BooksUnique).data({
+            ID: 1,
+            title: 'foo',
+            pages: [
+              {
+                ID: 5,
+                number: 1, // would fail without the update below first
+              },
+              {
+                ID: 3,
+                number: 999, // 1 -> 999
+              },
+            ],
+          }),
+        )
```

Comment:
With the amount of duplicate data being defined it is very difficult to identify what is the actual changes that are being applied. So I would go for an approach that re uses the original data to make it a bit easier to follow what the test is trying to do. While keeping the comments that explain the conflicts the data is trying to create / avoid.

```suggestion
        const data = {
              ID: 1,
              title: 'foo',
              pages: [
                {
                  ID: 1,
                  number: 1,
                },
                {
                  ID: 2,
                  number: 2,
                },
              ],
            }
        await tx.run(
          INSERT.into(BooksUnique).entries([ data ]),
        )
        
        // Create new entries with conflicting numbers
        data.pages[0].ID = 3
        data.pages[1].ID = 4
        await tx.run(UPDATE(BooksUnique).data(data)) // first, old entries are deleted, so no violation
        
        data.pages[0].ID = 5
        data.pages[0].number = 1  // would fail without the update below first
        data.pages[1].number = 999
        await tx.run(UPDATE(BooksUnique).data(data))
```

Reply:
I found the first one a bit more readable, but this change is fine for me